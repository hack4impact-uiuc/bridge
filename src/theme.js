/* eslint-disable prefer-destructuring */

function fontStack(fonts) {
  return fonts.map((font) => (font.includes(' ') ? `"${font}"` : font)).join(', ');
}

const DEFAULT_SHADOW = '0 1px 6px 2px rgba(0,0,0,0.15)';

const baseColors = {
  hackBlue: ['#B4D8FA', '#3E87CD', '#155DA1', '#0E4E8A', '#0D3F6E', '#031425'],
  indigoBlue: ['#BCD0F3', '#415F94', '#2D4979', '#253E68', '#13294E', '#061734'],
  impactGrey: ['#FAFBFC', '#EBEEF2', '#8B9199', '#666B72', '#46494F'],
};

const hackBluePalette = {
  lighter: baseColors.hackBlue[0],
  light: baseColors.hackBlue[1],
  primary: baseColors.hackBlue[2],
  dark: baseColors.hackBlue[3],
  darker: baseColors.hackBlue[4],
  text: baseColors.hackBlue[5],
};

const indigoPalette = {
  lighter: baseColors.indigoBlue[0],
  light: baseColors.indigoBlue[1],
  primary: baseColors.indigoBlue[2],
  dark: baseColors.indigoBlue[3],
  darker: baseColors.indigoBlue[4],
  text: baseColors.indigoBlue[5],
};

const impactGreyPalette = {
  lighter: baseColors.impactGrey[0],
  light: baseColors.impactGrey[1],
  midtone: baseColors.impactGrey[2],
  primary: baseColors.impactGrey[3],
  darker: baseColors.impactGrey[4],
};

const secondaryColors = {
  green: ['#BDF3C6', '#90E19E', '#4C9859', '#256830', '#154C1E', '#031D07'],
  red: ['#FFBCBF', '#D1595F', '#A62B31', '#8A1A1F', '#6B1014', '#35080A'],
  yellow: ['#FFECBD', '#FFD15F', '#FDBE21', '#DA9C03', '#A77700', '#201803'],
};

const greenPalette = {
  lighter: secondaryColors.green[0],
  light: secondaryColors.green[1],
  primary: secondaryColors.green[2],
  dark: secondaryColors.green[3],
  darker: secondaryColors.green[4],
  text: secondaryColors.green[5],
};

const redPalette = {
  lighter: secondaryColors.red[0],
  light: secondaryColors.red[1],
  primary: secondaryColors.red[2],
  dark: secondaryColors.red[3],
  darker: secondaryColors.red[4],
  text: secondaryColors.red[5],
};

const yellowPalette = {
  lighter: secondaryColors.yellow[0],
  light: secondaryColors.yellow[1],
  primary: secondaryColors.yellow[2],
  dark: secondaryColors.yellow[3],
  darker: secondaryColors.yellow[4],
  text: secondaryColors.yellow[5],
};

const typographyColors = {
  charcoal: '#272A2D',
  iron: '#4B4E52',
  ink: '#0A162A',
  slate: '#363949',
};

const colors = {
  black: '#000000',
  white: '#FFFFFF',
  ...baseColors,
  ...secondaryColors,
  ...typographyColors,
  hackBluePalette,
  indigoPalette,
  impactGreyPalette,
  greenPalette,
  redPalette,
  yellowPalette,
  text: {
    grayScale: {
      main: typographyColors.charcoal,
      support: typographyColors.iron,
    },
    coolTone: {
      main: typographyColors.ink,
      support: typographyColors.slate,
    },
  },
  // states or in other words color variants
  variants: {
    primary: hackBluePalette,
    success: greenPalette,
    error: redPalette,
    warning: yellowPalette,
    secondary: hackBluePalette,
    white: {
      primary: '#FFFFFF',
      dark: '#FFFFFF',
      lighter: indigoPalette.light,
    },
  },
};

const fontSizes = ['12px', '14px', '16px', '18px', '20px', '22px', '25px', '28px', '32px', '45px', '60px'];

const fonts = {
  chivo: fontStack(['Chivo']),
  hkGrotesk: fontStack(['HKGrotesk']),
};

const fontWeights = {
  light: 300,
  normal: 400,
  medium: 500,
  bold: 700,
};

// styling specific to the Button component
const button = {
  font: fonts.chivo,
  fontWeight: fontWeights.bold,
  fontSize: '14px',
  letterSpacing: '2px',
  // default values, based on the color palette
  default: (colorPalette) => ({
    bg: {
      default: colorPalette.primary,
      hover: colorPalette.dark,
      disabled: colorPalette.lighter,
    },
    border: {
      default: 'none',
      hover: 'none',
      disabled: 'none',
    },
    fontColor: {
      default: colors.white,
      hover: colors.white,
      disabled: colorPalette.light,
    },
    shadow: {
      default: 'none',
      hover: 'none',
      disabled: 'none',
    },
    outline: {
      bg: {
        default: colors.white,
        hover: colors.white,
        disabled: colors.white,
      },
      border: {
        default: colorPalette.primary,
        hover: colorPalette.dark,
        disabled: colorPalette.lighter,
      },
      fontColor: {
        default: colorPalette.primary,
        hover: colorPalette.dark,
        disabled: colorPalette.lighter,
      },
      shadow: {
        default: 'none',
        hover: 'none',
        disabled: 'none',
      },
    },
  }),

  // everything below overrides the default values
  primary: {
    fontColor: {
      default: colors.white,
      hover: colors.white,
      disabled: colors.white,
    },
  },
  white: {
    fontColor: {
      default: indigoPalette.primary,
      hover: indigoPalette.primary,
      disabled: indigoPalette.lighter,
    },
    shadow: {
      hover: DEFAULT_SHADOW,
    },
    outline: {
      bg: {
        default: indigoPalette.primary,
        hover: indigoPalette.primary,
        disabled: indigoPalette.primary,
      },
      border: {
        disabled: indigoPalette.light,
      },
      fontColor: {
        default: colors.white,
        hover: colors.white,
        disabled: indigoPalette.lighter,
      },
      shadow: {
        hover: DEFAULT_SHADOW,
      },
    },
  },
  secondary: {
    bg: {
      hover: hackBluePalette.darker,
      disabled: impactGreyPalette.light,
    },
    fontColor: {
      disabled: impactGreyPalette.midtone,
    },
    outline: {
      bg: {
        disabled: colors.white,
      },
      border: {
        hover: hackBluePalette.darker,
        disabled: impactGreyPalette.light,
      },
      fontColor: {
        disabled: indigoPalette.lighter,
      },
    },
  },
};


const link = {
  default: (colorPalette) => ({
    color: {
      default: colorPalette.primary,
      hover: colorPalette.dark,
    },
  }),
  secondary: {
    default: hackBluePalette.primary,
    hover: hackBluePalette.darker,
  },
};


const theme = {
  // General
  colors,
  fonts,
  fontWeights,
  fontSizes,
  button,
  link,
  DEFAULT_SHADOW,
};


export default theme;
export { colors };
