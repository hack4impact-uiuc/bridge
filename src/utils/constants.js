import * as styledSystem from 'styled-system';
import systemPropTypes from '@styled-system/prop-types';
import PropTypes from 'prop-types';


const { compose, system } = styledSystem;

// Text Decoration Prop
export const textDecoration = system({
  prop: 'textDecoration',
  cssProperty: 'text-decoration',
});
textDecoration.propTypes = PropTypes.string;


// common style props to be passed into components
export const COMMON = compose(styledSystem.space, styledSystem.layout);
COMMON.propTypes = {
  ...systemPropTypes.space,
  ...systemPropTypes.layout,
};

// for easier access
export const TYPOGRAPHY = styledSystem.typography;
export const LAYOUT = styledSystem.layout;
export const POSITION = styledSystem.position;
export const FLEX = styledSystem.flexbox;
export const GRID = styledSystem.grid;
export const COLOR = styledSystem.color;
export const BORDER = styledSystem.border;

TYPOGRAPHY.propTypes = systemPropTypes.typography;
LAYOUT.propTypes = systemPropTypes.layout;
POSITION.propTypes = systemPropTypes.position;
FLEX.propTypes = systemPropTypes.flexbox;
GRID.propTypes = systemPropTypes.grid;
COLOR.propTypes = systemPropTypes.color;
BORDER.propTypes = systemPropTypes.border;
