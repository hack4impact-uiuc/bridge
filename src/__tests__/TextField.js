import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { axe, toHaveNoViolations } from 'jest-axe';

import { TextField, Icon } from '..';
import { colors } from '../theme';
import { COLOR, COMMON } from '../utils/constants';
import { renderJSON } from '../utils/testing';
import 'babel-polyfill'; // axe violations required babel-polyfill

expect.extend(toHaveNoViolations);

describe('TextField', () => {
  it('renders a <div>', () => {
    expect(renderJSON(<TextField />).type).toEqual('div');
  });

  it('should have no axe violations', async () => {
    const { container } = render(<TextField aria-label="test" />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    cleanup();
  });

  it('implements system props', () => {
    expect(TextField).toImplementSystemProps(COLOR);
    expect(TextField).toImplementSystemProps(COMMON);
  });

  it('respects "width" props', () => {
    const { container } = render(<TextField width={200} />);
    expect(container.firstChild).toHaveStyle('width: 200px;');
  });

  it('respects the "disabled" prop', () => {
    const item = renderJSON(<TextField disabled />);
    expect(item.props.disabled).toEqual(true);
    expect(item).toMatchSnapshot();
  });

  it('respects "error" prop', () => {
    const item = renderJSON(<TextField error="error" />);
    expect(item).toMatchSnapshot();
  });

  it('renders correct colors with "disabled" props', () => {
    const { container } = render(<TextField disabled />);
    expect(container.firstChild).toHaveStyle(
      `color: ${colors.greyPalette.light}`,
    );
    expect(container.firstChild).toHaveStyle(
      `background-color: ${colors.greyPalette.lighter}`,
    );
    expect(container.firstChild).toHaveStyle(
      `border: 1px solid ${colors.greyPalette.light}`,
    );

    const item = renderJSON(<TextField disabled />);
    expect(item).toMatchSnapshot();
  });

  it('renders correct icon', () => {
    const item = renderJSON(<TextField icon={<Icon type="errorAlert" />} />);
    expect(item).toMatchSnapshot();
  });

  it('renders textarea using "as" prop', () => {
    const item = expect(renderJSON(<TextField as="textarea" />));
    expect(item).toMatchSnapshot();
  });
});
