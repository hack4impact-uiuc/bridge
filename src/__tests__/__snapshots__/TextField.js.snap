// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders correct colors with "disabled" props 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 46px;
  width: 100%;
  border: 1px solid #8B9199;
  border-radius: 4px;
  box-sizing: border-box;
  overflow: hidden;
  padding: 0px 9px 0px 9px;
}

.c0:focus-within {
  border: 2px solid #155DA1;
}

.c0[disabled] {
  background-color: #EBEEF2;
  border: 1px solid #8B9199;
  color: #8B9199;
}

.c1 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 18px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 20px;
  color: #0A162A;
  display: inline-block;
  width: 100%;
  outline: none;
  border: none;
  padding: 0;
  position: relative;
}

.c1:disabled {
  background-color: inherit;
  color: inherit;
}

<div
  className="c0"
  disabled={true}
>
  <input
    className="c1"
    disabled={true}
  />
</div>
`;

exports[`TextField renders correct icon 1`] = `
.c2 {
  display: inline-block;
  vertical-align: center;
  fill: none;
  width: 20px;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 46px;
  width: 100%;
  border: 1px solid #8B9199;
  border-radius: 4px;
  box-sizing: border-box;
  overflow: hidden;
  padding: 0px 9px 0px 9px;
}

.c0:focus-within {
  border: 2px solid #155DA1;
}

.c0[disabled] {
  background-color: #EBEEF2;
  border: 1px solid #8B9199;
  color: #8B9199;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 8px;
}

.c3 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 18px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 20px;
  color: #0A162A;
  display: inline-block;
  width: 100%;
  outline: none;
  border: none;
  padding: 0;
  position: relative;
}

.c3:disabled {
  background-color: inherit;
  color: inherit;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    <svg
      className="c2"
      fill="none"
      viewBox="0 0 30 30"
      width="20px"
    >
      <path
        clipRule="evenodd"
        d="M15 29C22.732 29 29 22.732 29 15C29 7.26801 22.732 1 15 1C7.26801 1 1 7.26801 1 15C1 22.732 7.26801 29 15 29Z"
        fillRule="evenodd"
        stroke="#A62B31"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
      <path
        d="M19.1998 10.8L10.7998 19.2"
        stroke="#A62B31"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
      <path
        d="M10.7998 10.8L19.1998 19.2"
        stroke="#A62B31"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </span>
  <input
    className="c3"
  />
</div>
`;

exports[`TextField renders textarea using "as" prop 1`] = `
Object {
  "lastCalledWith": [Function],
  "lastReturnedWith": [Function],
  "not": Object {
    "lastCalledWith": [Function],
    "lastReturnedWith": [Function],
    "nthCalledWith": [Function],
    "nthReturnedWith": [Function],
    "toBe": [Function],
    "toBeCalled": [Function],
    "toBeCalledTimes": [Function],
    "toBeCalledWith": [Function],
    "toBeChecked": [Function],
    "toBeCloseTo": [Function],
    "toBeDefined": [Function],
    "toBeDisabled": [Function],
    "toBeEmpty": [Function],
    "toBeEnabled": [Function],
    "toBeFalsy": [Function],
    "toBeGreaterThan": [Function],
    "toBeGreaterThanOrEqual": [Function],
    "toBeInTheDOM": [Function],
    "toBeInTheDocument": [Function],
    "toBeInstanceOf": [Function],
    "toBeInvalid": [Function],
    "toBeLessThan": [Function],
    "toBeLessThanOrEqual": [Function],
    "toBeNaN": [Function],
    "toBeNull": [Function],
    "toBeRequired": [Function],
    "toBeTruthy": [Function],
    "toBeUndefined": [Function],
    "toBeValid": [Function],
    "toBeVisible": [Function],
    "toContain": [Function],
    "toContainElement": [Function],
    "toContainEqual": [Function],
    "toContainHTML": [Function],
    "toEqual": [Function],
    "toHaveAttribute": [Function],
    "toHaveBeenCalled": [Function],
    "toHaveBeenCalledTimes": [Function],
    "toHaveBeenCalledWith": [Function],
    "toHaveBeenLastCalledWith": [Function],
    "toHaveBeenNthCalledWith": [Function],
    "toHaveClass": [Function],
    "toHaveDescription": [Function],
    "toHaveDisplayValue": [Function],
    "toHaveFocus": [Function],
    "toHaveFormValues": [Function],
    "toHaveLastReturnedWith": [Function],
    "toHaveLength": [Function],
    "toHaveNoViolations": [Function],
    "toHaveNthReturnedWith": [Function],
    "toHaveProperty": [Function],
    "toHaveReturned": [Function],
    "toHaveReturnedTimes": [Function],
    "toHaveReturnedWith": [Function],
    "toHaveStyle": [Function],
    "toHaveStyleRule": [Function],
    "toHaveTextContent": [Function],
    "toHaveValue": [Function],
    "toImplementSystemProps": [Function],
    "toMatch": [Function],
    "toMatchInlineSnapshot": [Function],
    "toMatchObject": [Function],
    "toMatchSnapshot": [Function],
    "toReturn": [Function],
    "toReturnTimes": [Function],
    "toReturnWith": [Function],
    "toStrictEqual": [Function],
    "toThrow": [Function],
    "toThrowError": [Function],
    "toThrowErrorMatchingInlineSnapshot": [Function],
    "toThrowErrorMatchingSnapshot": [Function],
  },
  "nthCalledWith": [Function],
  "nthReturnedWith": [Function],
  "rejects": Object {
    "lastCalledWith": [Function],
    "lastReturnedWith": [Function],
    "not": Object {
      "lastCalledWith": [Function],
      "lastReturnedWith": [Function],
      "nthCalledWith": [Function],
      "nthReturnedWith": [Function],
      "toBe": [Function],
      "toBeCalled": [Function],
      "toBeCalledTimes": [Function],
      "toBeCalledWith": [Function],
      "toBeChecked": [Function],
      "toBeCloseTo": [Function],
      "toBeDefined": [Function],
      "toBeDisabled": [Function],
      "toBeEmpty": [Function],
      "toBeEnabled": [Function],
      "toBeFalsy": [Function],
      "toBeGreaterThan": [Function],
      "toBeGreaterThanOrEqual": [Function],
      "toBeInTheDOM": [Function],
      "toBeInTheDocument": [Function],
      "toBeInstanceOf": [Function],
      "toBeInvalid": [Function],
      "toBeLessThan": [Function],
      "toBeLessThanOrEqual": [Function],
      "toBeNaN": [Function],
      "toBeNull": [Function],
      "toBeRequired": [Function],
      "toBeTruthy": [Function],
      "toBeUndefined": [Function],
      "toBeValid": [Function],
      "toBeVisible": [Function],
      "toContain": [Function],
      "toContainElement": [Function],
      "toContainEqual": [Function],
      "toContainHTML": [Function],
      "toEqual": [Function],
      "toHaveAttribute": [Function],
      "toHaveBeenCalled": [Function],
      "toHaveBeenCalledTimes": [Function],
      "toHaveBeenCalledWith": [Function],
      "toHaveBeenLastCalledWith": [Function],
      "toHaveBeenNthCalledWith": [Function],
      "toHaveClass": [Function],
      "toHaveDescription": [Function],
      "toHaveDisplayValue": [Function],
      "toHaveFocus": [Function],
      "toHaveFormValues": [Function],
      "toHaveLastReturnedWith": [Function],
      "toHaveLength": [Function],
      "toHaveNoViolations": [Function],
      "toHaveNthReturnedWith": [Function],
      "toHaveProperty": [Function],
      "toHaveReturned": [Function],
      "toHaveReturnedTimes": [Function],
      "toHaveReturnedWith": [Function],
      "toHaveStyle": [Function],
      "toHaveStyleRule": [Function],
      "toHaveTextContent": [Function],
      "toHaveValue": [Function],
      "toImplementSystemProps": [Function],
      "toMatch": [Function],
      "toMatchInlineSnapshot": [Function],
      "toMatchObject": [Function],
      "toMatchSnapshot": [Function],
      "toReturn": [Function],
      "toReturnTimes": [Function],
      "toReturnWith": [Function],
      "toStrictEqual": [Function],
      "toThrow": [Function],
      "toThrowError": [Function],
      "toThrowErrorMatchingInlineSnapshot": [Function],
      "toThrowErrorMatchingSnapshot": [Function],
    },
    "nthCalledWith": [Function],
    "nthReturnedWith": [Function],
    "toBe": [Function],
    "toBeCalled": [Function],
    "toBeCalledTimes": [Function],
    "toBeCalledWith": [Function],
    "toBeChecked": [Function],
    "toBeCloseTo": [Function],
    "toBeDefined": [Function],
    "toBeDisabled": [Function],
    "toBeEmpty": [Function],
    "toBeEnabled": [Function],
    "toBeFalsy": [Function],
    "toBeGreaterThan": [Function],
    "toBeGreaterThanOrEqual": [Function],
    "toBeInTheDOM": [Function],
    "toBeInTheDocument": [Function],
    "toBeInstanceOf": [Function],
    "toBeInvalid": [Function],
    "toBeLessThan": [Function],
    "toBeLessThanOrEqual": [Function],
    "toBeNaN": [Function],
    "toBeNull": [Function],
    "toBeRequired": [Function],
    "toBeTruthy": [Function],
    "toBeUndefined": [Function],
    "toBeValid": [Function],
    "toBeVisible": [Function],
    "toContain": [Function],
    "toContainElement": [Function],
    "toContainEqual": [Function],
    "toContainHTML": [Function],
    "toEqual": [Function],
    "toHaveAttribute": [Function],
    "toHaveBeenCalled": [Function],
    "toHaveBeenCalledTimes": [Function],
    "toHaveBeenCalledWith": [Function],
    "toHaveBeenLastCalledWith": [Function],
    "toHaveBeenNthCalledWith": [Function],
    "toHaveClass": [Function],
    "toHaveDescription": [Function],
    "toHaveDisplayValue": [Function],
    "toHaveFocus": [Function],
    "toHaveFormValues": [Function],
    "toHaveLastReturnedWith": [Function],
    "toHaveLength": [Function],
    "toHaveNoViolations": [Function],
    "toHaveNthReturnedWith": [Function],
    "toHaveProperty": [Function],
    "toHaveReturned": [Function],
    "toHaveReturnedTimes": [Function],
    "toHaveReturnedWith": [Function],
    "toHaveStyle": [Function],
    "toHaveStyleRule": [Function],
    "toHaveTextContent": [Function],
    "toHaveValue": [Function],
    "toImplementSystemProps": [Function],
    "toMatch": [Function],
    "toMatchInlineSnapshot": [Function],
    "toMatchObject": [Function],
    "toMatchSnapshot": [Function],
    "toReturn": [Function],
    "toReturnTimes": [Function],
    "toReturnWith": [Function],
    "toStrictEqual": [Function],
    "toThrow": [Function],
    "toThrowError": [Function],
    "toThrowErrorMatchingInlineSnapshot": [Function],
    "toThrowErrorMatchingSnapshot": [Function],
  },
  "resolves": Object {
    "lastCalledWith": [Function],
    "lastReturnedWith": [Function],
    "not": Object {
      "lastCalledWith": [Function],
      "lastReturnedWith": [Function],
      "nthCalledWith": [Function],
      "nthReturnedWith": [Function],
      "toBe": [Function],
      "toBeCalled": [Function],
      "toBeCalledTimes": [Function],
      "toBeCalledWith": [Function],
      "toBeChecked": [Function],
      "toBeCloseTo": [Function],
      "toBeDefined": [Function],
      "toBeDisabled": [Function],
      "toBeEmpty": [Function],
      "toBeEnabled": [Function],
      "toBeFalsy": [Function],
      "toBeGreaterThan": [Function],
      "toBeGreaterThanOrEqual": [Function],
      "toBeInTheDOM": [Function],
      "toBeInTheDocument": [Function],
      "toBeInstanceOf": [Function],
      "toBeInvalid": [Function],
      "toBeLessThan": [Function],
      "toBeLessThanOrEqual": [Function],
      "toBeNaN": [Function],
      "toBeNull": [Function],
      "toBeRequired": [Function],
      "toBeTruthy": [Function],
      "toBeUndefined": [Function],
      "toBeValid": [Function],
      "toBeVisible": [Function],
      "toContain": [Function],
      "toContainElement": [Function],
      "toContainEqual": [Function],
      "toContainHTML": [Function],
      "toEqual": [Function],
      "toHaveAttribute": [Function],
      "toHaveBeenCalled": [Function],
      "toHaveBeenCalledTimes": [Function],
      "toHaveBeenCalledWith": [Function],
      "toHaveBeenLastCalledWith": [Function],
      "toHaveBeenNthCalledWith": [Function],
      "toHaveClass": [Function],
      "toHaveDescription": [Function],
      "toHaveDisplayValue": [Function],
      "toHaveFocus": [Function],
      "toHaveFormValues": [Function],
      "toHaveLastReturnedWith": [Function],
      "toHaveLength": [Function],
      "toHaveNoViolations": [Function],
      "toHaveNthReturnedWith": [Function],
      "toHaveProperty": [Function],
      "toHaveReturned": [Function],
      "toHaveReturnedTimes": [Function],
      "toHaveReturnedWith": [Function],
      "toHaveStyle": [Function],
      "toHaveStyleRule": [Function],
      "toHaveTextContent": [Function],
      "toHaveValue": [Function],
      "toImplementSystemProps": [Function],
      "toMatch": [Function],
      "toMatchInlineSnapshot": [Function],
      "toMatchObject": [Function],
      "toMatchSnapshot": [Function],
      "toReturn": [Function],
      "toReturnTimes": [Function],
      "toReturnWith": [Function],
      "toStrictEqual": [Function],
      "toThrow": [Function],
      "toThrowError": [Function],
      "toThrowErrorMatchingInlineSnapshot": [Function],
      "toThrowErrorMatchingSnapshot": [Function],
    },
    "nthCalledWith": [Function],
    "nthReturnedWith": [Function],
    "toBe": [Function],
    "toBeCalled": [Function],
    "toBeCalledTimes": [Function],
    "toBeCalledWith": [Function],
    "toBeChecked": [Function],
    "toBeCloseTo": [Function],
    "toBeDefined": [Function],
    "toBeDisabled": [Function],
    "toBeEmpty": [Function],
    "toBeEnabled": [Function],
    "toBeFalsy": [Function],
    "toBeGreaterThan": [Function],
    "toBeGreaterThanOrEqual": [Function],
    "toBeInTheDOM": [Function],
    "toBeInTheDocument": [Function],
    "toBeInstanceOf": [Function],
    "toBeInvalid": [Function],
    "toBeLessThan": [Function],
    "toBeLessThanOrEqual": [Function],
    "toBeNaN": [Function],
    "toBeNull": [Function],
    "toBeRequired": [Function],
    "toBeTruthy": [Function],
    "toBeUndefined": [Function],
    "toBeValid": [Function],
    "toBeVisible": [Function],
    "toContain": [Function],
    "toContainElement": [Function],
    "toContainEqual": [Function],
    "toContainHTML": [Function],
    "toEqual": [Function],
    "toHaveAttribute": [Function],
    "toHaveBeenCalled": [Function],
    "toHaveBeenCalledTimes": [Function],
    "toHaveBeenCalledWith": [Function],
    "toHaveBeenLastCalledWith": [Function],
    "toHaveBeenNthCalledWith": [Function],
    "toHaveClass": [Function],
    "toHaveDescription": [Function],
    "toHaveDisplayValue": [Function],
    "toHaveFocus": [Function],
    "toHaveFormValues": [Function],
    "toHaveLastReturnedWith": [Function],
    "toHaveLength": [Function],
    "toHaveNoViolations": [Function],
    "toHaveNthReturnedWith": [Function],
    "toHaveProperty": [Function],
    "toHaveReturned": [Function],
    "toHaveReturnedTimes": [Function],
    "toHaveReturnedWith": [Function],
    "toHaveStyle": [Function],
    "toHaveStyleRule": [Function],
    "toHaveTextContent": [Function],
    "toHaveValue": [Function],
    "toImplementSystemProps": [Function],
    "toMatch": [Function],
    "toMatchInlineSnapshot": [Function],
    "toMatchObject": [Function],
    "toMatchSnapshot": [Function],
    "toReturn": [Function],
    "toReturnTimes": [Function],
    "toReturnWith": [Function],
    "toStrictEqual": [Function],
    "toThrow": [Function],
    "toThrowError": [Function],
    "toThrowErrorMatchingInlineSnapshot": [Function],
    "toThrowErrorMatchingSnapshot": [Function],
  },
  "toBe": [Function],
  "toBeCalled": [Function],
  "toBeCalledTimes": [Function],
  "toBeCalledWith": [Function],
  "toBeChecked": [Function],
  "toBeCloseTo": [Function],
  "toBeDefined": [Function],
  "toBeDisabled": [Function],
  "toBeEmpty": [Function],
  "toBeEnabled": [Function],
  "toBeFalsy": [Function],
  "toBeGreaterThan": [Function],
  "toBeGreaterThanOrEqual": [Function],
  "toBeInTheDOM": [Function],
  "toBeInTheDocument": [Function],
  "toBeInstanceOf": [Function],
  "toBeInvalid": [Function],
  "toBeLessThan": [Function],
  "toBeLessThanOrEqual": [Function],
  "toBeNaN": [Function],
  "toBeNull": [Function],
  "toBeRequired": [Function],
  "toBeTruthy": [Function],
  "toBeUndefined": [Function],
  "toBeValid": [Function],
  "toBeVisible": [Function],
  "toContain": [Function],
  "toContainElement": [Function],
  "toContainEqual": [Function],
  "toContainHTML": [Function],
  "toEqual": [Function],
  "toHaveAttribute": [Function],
  "toHaveBeenCalled": [Function],
  "toHaveBeenCalledTimes": [Function],
  "toHaveBeenCalledWith": [Function],
  "toHaveBeenLastCalledWith": [Function],
  "toHaveBeenNthCalledWith": [Function],
  "toHaveClass": [Function],
  "toHaveDescription": [Function],
  "toHaveDisplayValue": [Function],
  "toHaveFocus": [Function],
  "toHaveFormValues": [Function],
  "toHaveLastReturnedWith": [Function],
  "toHaveLength": [Function],
  "toHaveNoViolations": [Function],
  "toHaveNthReturnedWith": [Function],
  "toHaveProperty": [Function],
  "toHaveReturned": [Function],
  "toHaveReturnedTimes": [Function],
  "toHaveReturnedWith": [Function],
  "toHaveStyle": [Function],
  "toHaveStyleRule": [Function],
  "toHaveTextContent": [Function],
  "toHaveValue": [Function],
  "toImplementSystemProps": [Function],
  "toMatch": [Function],
  "toMatchInlineSnapshot": [Function],
  "toMatchObject": [Function],
  "toMatchSnapshot": [Function],
  "toReturn": [Function],
  "toReturnTimes": [Function],
  "toReturnWith": [Function],
  "toStrictEqual": [Function],
  "toThrow": [Function],
  "toThrowError": [Function],
  "toThrowErrorMatchingInlineSnapshot": [Function],
  "toThrowErrorMatchingSnapshot": [Function],
}
`;

exports[`TextField respects "error" prop 1`] = `
.c2 {
  display: inline-block;
  vertical-align: center;
  fill: none;
  width: 20px;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 46px;
  width: 100%;
  border: 1px solid #A62B31;
  border-radius: 4px;
  box-sizing: border-box;
  overflow: hidden;
  padding: 0px 9px 0px 9px;
}

.c0:focus-within {
  border: 2px solid #A62B31;
}

.c0[disabled] {
  background-color: #EBEEF2;
  border: 1px solid #8B9199;
  color: #8B9199;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 4px;
  width: 16px;
}

.c1 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 18px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 20px;
  color: #0A162A;
  display: inline-block;
  width: 100%;
  outline: none;
  border: none;
  padding: 0;
  position: relative;
}

.c1:disabled {
  background-color: inherit;
  color: inherit;
}

<div
  className="c0"
>
  <input
    className="c1"
  />
  <svg
    className="c2 c3"
    fill="none"
    viewBox="0 0 18 18"
    width="20px"
  >
    <path
      clipRule="evenodd"
      d="M9 17C13.4183 17 17 13.4183 17 9C17 4.58172 13.4183 1 9 1C4.58172 1 1 4.58172 1 9C1 13.4183 4.58172 17 9 17Z"
      fillRule="evenodd"
      stroke="#A62B31"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1.25"
    />
    <path
      d="M8.99961 5.7998V8.9998"
      stroke="#A62B31"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1.25"
    />
    <path
      d="M8.99952 12.1998H9.00752"
      stroke="#A62B31"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1.25"
    />
  </svg>
</div>
`;

exports[`TextField respects the "disabled" prop 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 46px;
  width: 100%;
  border: 1px solid #8B9199;
  border-radius: 4px;
  box-sizing: border-box;
  overflow: hidden;
  padding: 0px 9px 0px 9px;
}

.c0:focus-within {
  border: 2px solid #155DA1;
}

.c0[disabled] {
  background-color: #EBEEF2;
  border: 1px solid #8B9199;
  color: #8B9199;
}

.c1 {
  font-family: HKGrotesk;
  font-weight: 400px;
  font-size: 18px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 20px;
  color: #0A162A;
  display: inline-block;
  width: 100%;
  outline: none;
  border: none;
  padding: 0;
  position: relative;
}

.c1:disabled {
  background-color: inherit;
  color: inherit;
}

<div
  className="c0"
  disabled={true}
>
  <input
    className="c1"
    disabled={true}
  />
</div>
`;
